openstates/importers/base.py

```diff
@@ -180,7 +181,24 @@ def resolve_bill(self, bill_id: str, *, date: str) -> typing.Optional[_ID]:
         if len(ids) == 1:
             return ids.pop()
         elif len(ids) == 0:
```

Comment:
Why only try the adjusted date on fallback query? If the main query is not doing the intended job (matching on greater-than-or-equal to same date), would it make sense to just bump the main query's date value down a day?

Reply:
Good point

Reply:
My thought here was to preserve the status  quo because it seems to be working in most of our use cases across jurisdiction. We don't want to start pushing the date on the first try and potentially returning 2 rows(which is also a failure). For example 
`2024 regular session ends 12-01-2024`
`2024 special  session starts 12-02-2024`
they both have HB 1. An event on the 12-02-2024 might return 2 rows?
So my thought was do the normal, if it fails(the rear case) then use the fall back query with an adjusted date.

Reply:
I hear you, it just "smells" a little funny to me: can't we get the primary query to perform the intended date cut off? In the case you mention (reg 12-01-24/special 12-02-24), wouldn't the fallback query get performed and return 2 rows anyway?

FWIW, we have had performance issues rise previously around the query in this function being called a bunch of times (since it gets called per-bill, per -scrape). So there is potentially mild-to-moderate real world impact to regularly making two queries instead of one (bills that don't get resolved tend to not get resolved for a long time, so they keep generating queries over time).

I'm not super close to this code/data, so I'm not trying to raise a blocking concern. If this is the best solution go with it, I'm just providing my perspective.

Reply:
>  In the case you mention (reg 12-01-24/special 12-02-24) , wouldn't the fallback query get performed and return 2 rows anyway?

So my thought is 
* perform the existing query. No Date change. I believe will work most of the time. Return 1 row. and then skip fall back query. Continue processing
* If existing query (No Date change) return 0 rows `elif len(ids) == 0:`. That is when we try again with adjusted date. I believe this happens less frequently
So we only run fall back query if there was 0 zero.

@NewAgeAirbender  correct me otherwise.  
Also if we are going to make the initial query `date - timedelta(days=1)` or `date - timedelta(days=7)` then I don't think there is a case for a fallback query. My suggestion for using a fallback query instead of the initial query for adjusted date was to avoid creating a new bug/issue in other jurisdictions while trying to solve this problem in AK and similar use case.